rsocket-core/src/test/java/io/rsocket/buffer/LeaksTrackingByteBufAllocator.java

```diff
@@ -0,0 +1,192 @@
+package io.rsocket.buffer;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.CompositeByteBuf;
+import org.assertj.core.api.Assertions;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+public class LeaksTrackingByteBufAllocator implements ByteBufAllocator {
+
+    public static LeaksTrackingByteBufAllocator instrumentDefault() {
+        if (ByteBufAllocator.DEFAULT instanceof LeaksTrackingByteBufAllocator) {
+            return (LeaksTrackingByteBufAllocator) ByteBufAllocator.DEFAULT;
+        }
+
+        final LeaksTrackingByteBufAllocator instrumented = new LeaksTrackingByteBufAllocator(ByteBufAllocator.DEFAULT);
+        try {
+            Field field = ByteBufAllocator.class.getField("DEFAULT");
+            setFinalStatic(field, instrumented);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        return instrumented;
+    }
```

Comment:
Maybe add implementation comment that this is temporary/necessary until https://github.com/rsocket/rsocket-java/issues/776 is resolved?

Reply:
Actually, I was planning to make that as instrumentation for testing to always check that we don't have any leaks

Reply:
Right I guess the question is where the `ByteBufAllocator` will come from and how it will be instrumented. Hopefully not by changing `ByteBufAllocator.DEFAULT` via reflection but yeah for now it's perfectly fine.